src/httpserver.cpp

```diff
@@ -207,6 +213,18 @@ std::string RequestMethodString(HTTPRequest::RequestMethod m)
 /** HTTP request callback */
 static void http_request_cb(struct evhttp_request* req, void* arg)
 {
+    // Track requests and notify when last active request is completed.
+    {
+        LOCK(g_requests_mutex);
+        g_requests.insert(req);
+        evhttp_request_set_on_complete_cb(req, [](struct evhttp_request* req, void*) {
+            LOCK(g_requests_mutex);
+            auto n = g_requests.erase(req);
+            assert(n == 1);
+            if (g_requests.empty()) g_requests_cv.notify_all();
```

Comment:
maybe always call `g_requests_cv.notify_all()` and update comment above.

Reply:
why would we want to always call `.notify_all()` if we are waiting for all the http requests to finish?

Reply:
because that way you would get a log every time a connection closes. Otherwise `g_requests_cv` should be renamed to `g_requests_empty_cv`.

If `g_requests_cv` is used to signal changes to `g_requests` then also add call `.notify_all()` after `g_requests.insert(req)`.

Reply:
done, I have opted for always calling it